lib/image.php

```diff
@@ -260,6 +260,26 @@ function __errorHandler($errno=NULL, $errstr, $errfile=NULL, $errline=NULL, $err
 		$etag = NULL;
 	}
 
+	// Allow CORS
+	// respond to preflights
+	if ($settings['image']['allow_origin'] !== null) {
+		if ($_SERVER['REQUEST_METHOD'] == 'OPTIONS') {
+			// return only the headers and not the content
+			// only allow CORS if we're doing a GET - i.e. no sending for now.
+			if (isset($_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD']) && $_SERVER['HTTP_ACCESS_CONTROL_REQUEST_METHOD'] == 'GET') {
+				header('Access-Control-Allow-Origin: *');
+				header('Access-Control-Allow-Headers: X-Requested-With');
+			}
+			exit;
+		} else {
+			header('Origin: ' . $settings['image']['allow_origin']);
+			header('Access-Control-Allow-Origin: ' . $settings['image']['allow_origin']);
+			header('Access-Control-Allow-Methods: GET');
+			header('Access-Control-Max-Age: 3000');
+			header('Vary: Origin, Access-Control-Request-Headers, Access-Control-Request-Method');
```

Comment:
Theorically, Vary is nice. If you control the stack (like a proxy in a CDN) it can be nice. But with browsers, it's a mess.